Library/Homebrew/command.rb

```diff
@@ -72,13 +72,51 @@ def self.desc(desc)
       end
     end
 
+    def self.get_invalid_options(argv_options_only)
+      argv_options_only = argv_options_only.uniq
+      valid_options_with_values =
+        @valid_options
+        .select { |option_hash| !option_hash[:value].nil? }
+        .map { |option_hash| option_hash[:option] }
+      all_valid_options =
+        @valid_options
+        .map { |option_hash| option_hash[:option] }
+
+      argv_invalid_options = []
+      argv_options_without_value = {}
+      argv_options_only.each do |argv_option|
+        if argv_option.include?("=")
+          index = argv_option.index("=")
+          argv_option_name = argv_option[0..index-1]
+          if !all_valid_options.include?(argv_option_name)
+            argv_invalid_options.push(argv_option_name)
+          elsif index == argv_option.size-1
+            option_value = @valid_options.find { |x| x[:option] == argv_option_name }[:value]
+            argv_options_without_value[argv_option_name] = option_value
+          end
+        elsif !all_valid_options.include?(argv_option)
+          argv_invalid_options.push(argv_option)
+        elsif valid_options_with_values.include?(argv_option)
+          option_value = @valid_options.find { |x| x[:option] == argv_option }[:value]
+          argv_options_without_value[argv_option] = option_value
+        end
+      end
+      [argv_invalid_options, argv_options_without_value]
+    end
+
     def self.get_error_message(argv_options_only)
```

Comment:
Similarly, it might make sense to break the logic in here into two different different methods that produce the error messages for invalid options and missing values respectively.